pyro/infer/predictive.py

```diff
@@ -219,3 +220,71 @@ def get_vectorized_trace(self, *args, **kwargs):
                                             parallel=self.parallel, model_args=args, model_kwargs=kwargs)
         return _predictive(self.model, posterior_samples, self.num_samples,
                            return_trace=True, model_args=args, model_kwargs=kwargs)
+
+
+def log_likelihood(
+    model: Callable,
+    posterior_samples: Dict[str, torch.Tensor] = None,
+    guide: Callable = None,
+    num_samples: int = None,
+    parallel: bool = True,
+) -> Callable:
+    """Calculates the log likelihood of the data for a model with observed
```

Comment:
Could you add an EXPERIMENTAL flag here? (like in [this docs](http://docs.pyro.ai/en/stable/primitives.html#pyro.primitives.deterministic))